Realm/ObjectServerTests/SwiftCollectionSyncTests.swift

```diff
@@ -338,3 +338,602 @@ class CollectionSyncTestCase: SwiftSyncTestCase {
+            .bool(false),
+            .int(234),
+            .double(12345.678901),
+            .float(12.34),
+            .data(Data("a".utf8)),
+            .date(date),
+            .object(so),
+            .objectId(oid),
+            .uuid(uuid),
+            .decimal128(Decimal128(number: 567))
+        ]
+    }
+
+    private func assertListEqual(_ object: SwiftTypesSyncObject, _ index: Int, _ value: AnyRealmValue) {
+        if case .object = value {
+            XCTAssertEqual(object.anyCol.listValue?[index].object(SwiftPerson.self)?.firstName, value.object(SwiftPerson.self)?.firstName)
+        } else if case .float = value {
+            if object.anyCol.listValue?[index].floatValue != nil {
+                XCTAssertEqual(object.anyCol.listValue?[index], value)
+            } else {
+                // This happens when we download a fresh realm
+                XCTAssertEqual(object.anyCol.listValue?[index], .double(Double(value.floatValue!)))
+            }
+        } else if case .date = value {
+            // Date is not exact when synced
+            let date = object.anyCol.listValue?[index].dateValue
+            XCTAssertTrue(Calendar.current.isDate(date!, equalTo: value.dateValue!, toGranularity: .second))
+        } else {
+            XCTAssertEqual(object.anyCol.listValue?[index], value)
+        }
+    }
+
+    private func assertDictionaryEqual(_ object: SwiftTypesSyncObject, _ key: String, _ value: AnyRealmValue) {
+        if case .object = value {
+            XCTAssertEqual(object.anyCol.dictionaryValue?[key]?.object(SwiftPerson.self)?.firstName, value.object(SwiftPerson.self)?.firstName)
+        } else if case .float = value {
+            if object.anyCol.dictionaryValue?[key]?.floatValue != nil {
+                XCTAssertEqual(object.anyCol.dictionaryValue?[key], value)
+            } else {
+                // This happens when we download a fresh realm
+                XCTAssertEqual(object.anyCol.dictionaryValue?[key], .double(Double(value.floatValue!)))
+            }
+        } else if case .date = value {
+            // Date is not exact when synced
+            let date = object.anyCol.dictionaryValue?[key]?.dateValue
+            XCTAssertTrue(Calendar.current.isDate(date!, equalTo: value.dateValue!, toGranularity: .second))
+        } else {
+            XCTAssertEqual(object.anyCol.dictionaryValue?[key], value)
+        }
+    }
+
+    @MainActor func testSyncAnyRealmValue() async throws {
+        let list = sampleData
+        for (index, value) in list.enumerated() {
+            try await write { realm in
+                let object = SwiftTypesSyncObject()
+                object.anyCol = value
+                object.stringCol = "\(self.name)_\(index)"
+                realm.add(object)
+            }
+
+            let realm = try await openRealm()
+            let results0 =  try await realm.objects(SwiftTypesSyncObject.self).where { $0.stringCol == "\(self.name)_\(index)" }.subscribe()
```

Comment:
None of the `.subscribe()`s in any of these tests do anything as the initial subscriptions are subscribing to all objects already.